core/src/test/java/mediumtest/TelemetryMediumTests.java

```diff
@@ -0,0 +1,117 @@
+    }
+  }
+
+  @Test
+  void it_should_not_create_telemetry_file_if_telemetry_disabled_by_system_property(@TempDir Path sonarlintUserHome) throws ExecutionException, InterruptedException {
+    System.setProperty("sonarlint.telemetry.disabled", "true");
+    var fakeClient = newFakeClient().build();
+    backend = newBackend()
+      .withSonarLintUserHome(sonarlintUserHome)
+      .withSonarQubeConnection("connectionId", "http://localhost:12345")
+      .withBoundConfigScope("scopeId", "connectionId", "projectKey")
+      .build(fakeClient);
+
+    assertThat(backend.getTelemetryService().getStatus().get().isEnabled()).isFalse();
+
+    this.backend.getHotspotService().openHotspotInBrowser(new OpenHotspotInBrowserParams("scopeId", "master", "ab12ef45"));
+    assertThat(sonarlintUserHome).isEmptyDirectory();
+  }
+
+  @Test
+  void it_should_create_telemetry_file_if_telemetry_enabled(@TempDir Path sonarlintUserHome) throws ExecutionException, InterruptedException {
+    System.clearProperty("sonarlint.telemetry.disabled");
+    var fakeClient = newFakeClient().build();
+    backend = newBackend()
+      .withSonarLintUserHome(sonarlintUserHome)
+      .withSonarQubeConnection("connectionId", "http://localhost:12345")
+      .withBoundConfigScope("scopeId", "connectionId", "projectKey")
+      .build(fakeClient);
+
+    assertThat(backend.getTelemetryService().getStatus().get().isEnabled()).isTrue();
+
+    this.backend.getHotspotService().openHotspotInBrowser(new OpenHotspotInBrowserParams("scopeId", "master", "ab12ef45"));
+    assertThat(sonarlintUserHome.resolve("telemetry/mediumTests/usage")).isNotEmptyFile();
+  }
+
+  @Test
+  void it_should_consider_telemetry_status_in_file(@TempDir Path sonarlintUserHome) throws ExecutionException, InterruptedException {
+    System.clearProperty("sonarlint.telemetry.disabled");
+    var fakeClient = newFakeClient().build();
+    backend = newBackend()
+      .withSonarLintUserHome(sonarlintUserHome)
+      .withSonarQubeConnection("connectionId", "http://localhost:12345")
+      .withBoundConfigScope("scopeId", "connectionId", "projectKey")
+      .build(fakeClient);
+
+    assertThat(backend.getTelemetryService().getStatus().get().isEnabled()).isTrue();
+
+    // Trigger any telemetry event to initialize the file
+    this.backend.getHotspotService().openHotspotInBrowser(new OpenHotspotInBrowserParams("scopeId", "master", "ab12ef45"));
+    assertThat(sonarlintUserHome.resolve("telemetry/mediumTests/usage")).isNotEmptyFile();
+
+    // Emulate another process has disabled telemetry
+    var telemetryLocalStorageManager = new TelemetryLocalStorageManager(TelemetryPathManager.getPath(sonarlintUserHome, SonarLintBackendFixture.MEDIUM_TESTS_PRODUCT_KEY));
+    telemetryLocalStorageManager.tryUpdateAtomically(data -> {
+      data.setEnabled(false);
+    });
+
+    assertThat(backend.getTelemetryService().getStatus().get().isEnabled()).isFalse();
+  }
+
```

Comment:
as it is mentioned in the Javadoc somewhere, we should maybe also test the default `~/.sonarlint` location is used if no SL home is provided

Reply:
I don't want to take the risk to mess up my local `~/.sonarlint`, but maybe I can add a unit test.